src/main/java/ca/corefacility/bioinformatics/irida/repositories/sample/SampleRepositoryImpl.java

```diff
@@ -0,0 +1,39 @@
+package ca.corefacility.bioinformatics.irida.repositories.sample;
+
+import ca.corefacility.bioinformatics.irida.model.project.Project;
+import ca.corefacility.bioinformatics.irida.model.sample.Sample;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+
+import javax.sql.DataSource;
+import java.util.List;
+
+/**
+ * Impl of custom methods for {@link SampleRepository}.  This class can be used for speed improvements for sample
+ * listing methods.
+ */
+public class SampleRepositoryImpl implements SampleRepositoryCustom {
+	private final DataSource dataSource;
+
+	@Autowired
+	public SampleRepositoryImpl(DataSource dataSource) {
+		this.dataSource = dataSource;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public List<Sample> getSamplesForProjectShallow(Project project) {
+		NamedParameterJdbcTemplate tmpl = new NamedParameterJdbcTemplate(dataSource);
+		MapSqlParameterSource parameters = new MapSqlParameterSource();
+
+		//query to read samples for a project
+		String queryString = "select s.id, s.createdDate, s.modifiedDate, s.description, s.sampleName, s.collectedBy, s.geographicLocationName, s.isolate, s.isolationSource, s.latitude, s.longitude, s.organism, s.strain, s.collectionDate, null as remote_status FROM sample s INNER JOIN project_sample p ON p.sample_id=s.id WHERE p.project_id=:project";
+
+		parameters.addValue("project", project.getId());
+
+		return tmpl.query(queryString, parameters, new BeanPropertyRowMapper(Sample.class));
```

Comment:
It looks like the `createdDate` for a sample is not being set correctly in these results. It's instead being set to today's date.

## Results from API for current IRIDA release

```
```
{
  "resource" : {
    "resources" : [ {
      "createdDate" : 1544652180042,
      "modifiedDate" : 1538512517000,
      "sampleName" : "08-5578",
...
```

The `createdDate` corresponds to today's date **Wed, 12 Dec 2018 22:03:00 +0000**

## Results from API from this code

```
{
  "resource" : {
    "resources" : [ {
      "createdDate" : 1535122022000,
      "modifiedDate" : 1538512517000,
      "sampleName" : "08-5578",
```

The `createdDate` corresponds to the correct sample date **Fri, 24 Aug 2018 14:47:02 +0000**

Reply:
Looks like the `BeanPropertyRowMapper` wasn't properly overriding the `createdDate` from the constructor.  Switched to using a manual `RowMapper` in ffc28e0.  Still just as fast.

Reply:
Awesome, thanks :+1: